src/contract.rs

```diff
@@ -0,0 +1,86 @@
+#[cfg(not(feature = "imported"))]
+use cosmwasm_std::entry_point;
+use cosmwasm_std::{DepsMut, Env, MessageInfo, Reply, Response};
+use cw2::set_contract_version;
+
+use crate::error::ContractError;
+use crate::execute::{handle_swap_reply, single_sided_swap_and_lp};
+use crate::msg::{ExecuteMsg, InstantiateMsg, MigrateMsg};
+use crate::state::SWAP_REPLY_STATES;
+use crate::state::{Config, CONFIG};
+
+// Msg Reply IDs
+pub const SWAP_REPLY_ID: u64 = 1u64;
+
+// version info for migration info
+const CONTRACT_NAME: &str = "crates.io:osmosis-single-sided-swap-and-lp";
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+/// Handling contract instantiation
+#[cfg_attr(not(feature = "imported"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut,
+    _env: Env,
+    _info: MessageInfo,
+    msg: InstantiateMsg,
+) -> Result<Response, ContractError> {
+    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
+
+    let owner = deps.api.addr_validate(&msg.owner)?;
+    let state = Config { owner };
+
+    CONFIG.save(deps.storage, &state)?;
+    Ok(Response::new().add_attribute("method", "instantiate"))
+}
```

Comment:
Currently contract owner doesn't have any special powers. I do think we should add the min tick, max tick, min spot price, max spot price constants as store values that can be changed by the contract owner, but that can be in the next iteration.